lib/global/backend/gcp_secret_manager.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+module Global
+  module Backend
+
+    # Loads Global configuration from the Google Cloud Secret Manager
+    # https://cloud.google.com/secret-manager/docs
+    #
+    # This backend requires the `google-cloud-secret_manager` gem, so make sure to add it to your Gemfile.
+    #
+    # Available options:
+    # - `project_id` (required): Google Cloud project name
+    # - `prefix` (required): the prefix in Secret Manager; all parameters within the prefix will be loaded;
+    #   make sure to add a undescore, if you want it
+    #   see https://cloud.google.com/secret-manager/docs/overview
+    # - `client`: pass you own Google::Cloud::SecretManager instance, or alternatively set:
+    # - `gcp_options`: credentials and other Google cloud configuration options
+    #   that are passed to Google::Cloud::SecretManager.configure
+    #   see https://googleapis.dev/ruby/google-cloud-secret_manager/latest/index.html
+    #   If Google Cloud access is already configured through environment variables,
+    #   you don't need to pass the credentials explicitly.
+    #
+    # For Rails:
+    # - the `prefix` is optional and defaults to `[Rails enviroment]_[Name of the app class]_`,
+    #    for example: `production_myapp_`
+    # - to use a different app name, pass `app_name`,
+    #    for example: `backend :gcp_secret_manager, app_name: 'new_name_for_my_app'`
+    class GcpSecretManager
+
+      GCP_SEPARATOR = '/'
+      PATH_SEPARATOR = '_'
+
+      def initialize(options = {})
+        @project_id = options.fetch(:project_id)
+        require_gcp_gem
+        init_prefix(options)
+        init_client(options)
+      end
+
+      def load
+        pages = load_all_parameters_from_gcsm
+
+        configuration = {}
+        pages.each do |page|
+          configuration.deep_merge!(build_configuration_from_pages(page))
+        end
+      end
+
+      private
+
+      def require_gcp_gem
+        require 'google/cloud/secret_manager'
+      rescue LoadError
+        raise 'The `google-cloud-secret_manager` gem must be installed.'
+      end
+
+      def init_prefix(options)
+        if defined?(Rails)
+          environment = Rails.env.to_s
```

Comment:
should you move this into the `fetch` fallback block?

Reply:
@leonid-shevtsov fixed